### ISSUE TITLE
SonarCSS freezes because stylelint freezes when parsing html.erb file

### ISSUE THREAD

**Issue by alban-auzeill** [SonarSource/sonar-css#222]
When the following file is analyzed by `stylelint` with the rule `string-no-newline`, `stylelint` freezes:
https://github.com/rubocop-hq/rubocop/blob/master/assets/output.html.erb

Smaller reproducer, given:
the file `config.json`
```json
{"rules": {"string-no-newline": true}}
```
the file `file.css`
```css
<% COLORS.each do |color| %>
a { }
<% end %>
a { }
```
the following command freezes:
```sh
stylelint --config config.json file.css
```

stylelint rule implementation:
https://github.com/stylelint/stylelint/tree/master/lib/rules/string-no-newline

**Comment by vilchik-elena** [comment]
reported to stylelint https://github.com/stylelint/stylelint/issues/4489

**Comment by vilchik-elena** [comment]
Also `no-descending-specificity` fails the same way on the same file. Note that even if 12.0.1 is still failing for this rule, master branch is ok (due to https://github.com/stylelint/stylelint/commit/0f04539ba1fe1639cbdfe6ba597d6676df2b4019). So let's wait for it to be released.
